storage/myisam/mi_create.c

```diff
@@ -710,16 +710,46 @@ int mi_create(const char *name,uint keys,MI_KEYDEF *keydefs,
   }
 #endif
 
+  k=0;
+  u=0;
+   offset= real_reclength - uniques * MI_UNIQUE_HASH_LENGTH;
+  bzero((char*) &tmp_keydef,sizeof(tmp_keydef));
+  bzero((char*) &tmp_keyseg,sizeof(tmp_keyseg));
   /* Write key and keyseg definitions */
   DBUG_PRINT("info", ("write key and keyseg definitions"));
-  for (i=0 ; i < share.base.keys - uniques; i++)
+  for (i=0 ; i < share.base.keys ; i++)
   {
-    uint sp_segs=(keydefs[i].flag & HA_SPATIAL) ? 2*SPDIMS : 0;
+    for (u=0, uniquedef=uniquedefs ; u < uniques ; u++ , uniquedef++)
+    {
```

Comment:
Do you really need u why not for(uniquedef=uniquedefs; uniquedef < uniques ; uniquedef++) ?